distributed/client.py

```diff
@@ -328,6 +328,10 @@ def done_callback(future, callback):
 def normalize_future(f):
     return [f.key, type(f)]
 
+def in_ioloop():
+    """ Are we running within the current IOLoop? """
+    return IOLoop.current()._running and IOLoop.current()._thread_ident == get_thread_identity()
```

Comment:
@kszucs @ajdavis @pitrou is there a better way to do this?  Is this generally possible?